src/App.tsx

```diff
@@ -1,26 +1,244 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useState } from 'react';
+import { UserWarning } from './components/UserWarning';
+import {
+  getTodos,
+  postTodo,
+  removeTodo,
+  renewTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorComponent } from './components/ErrorComponent';
+import { Footer } from './components/Footer';
+import { getFilteredItems } from './utils/getFilteredItems';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Filter } from './utils/enamFilter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState<'all' | 'active' | 'completed'>(
+    Filter.All,
+  );
+  const [error, setError] = useState('');
+  const [title, setTitle] = useState('');
+  const [loading, setLoading] = useState<number | null>(null);
+  const [isEditingId, setIsEditingId] = useState<number | null>(null);
+  const showError = (message: string) => {
+    setError(message);
+
+    setTimeout(() => {
+      setError('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => showError('Unable to load todos'))
+      .finally(() => setLoading(0));
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const addTodo = (todoTitle: string) => {
+    setLoading(1);
+    setTempTodo({
+      id: 0,
+      userId: 2215,
+      title: todoTitle,
+      completed: false,
+    });
+
+    postTodo(title.trim())
+      .then((newTodo: Todo) => {
+        setTodos(prevTodos => [...prevTodos, newTodo]);
+        setError('');
+        setTitle('');
+      })
+      .catch(() => showError('Unable to add a todo'))
+      .finally(() => {
+        setLoading(0);
+        setTempTodo(null);
+      });
+  };
+
+  const updateTodoCheck = (id: number) => {
+    setLoading(id);
+
+    const todoToUpdate = todos.find(todo => todo.id === id);
+
+    if (!todoToUpdate) {
+      setLoading(0);
+
+      return;
+    }
+
+    const updatedTodo = {
+      ...todoToUpdate,
+      completed: !todoToUpdate.completed,
+    };
+
+    renewTodo(id, updatedTodo)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.map(todo =>
+            todo.id === id ? { ...todo, completed: !todo.completed } : todo,
+          ),
+        );
+        setLoading(0);
+      })
+      .catch(() => showError('Unable to update a todo'))
+      .finally(() => setLoading(0));
+  };
+
+  function updateTodoTitle(id: number, value: string): Promise<void> | void {
```

Comment:
It is important to follow the same way of creating functions everywhere, fix it everywhere
```suggestion
  const updateTodoTitle = (id: number, value: string): Promise<void> | void => {
```